docs/features/string-literals-data-section.md

```diff
@@ -0,0 +1,201 @@
+# String literals in data section
+
+This opt-in **experimental** Roslyn feature allows changing how string literals in C# programs are emitted into PE files (`.dll`/`.exe`).
+By default, string literals are emitted to the UserString heap which is limited to [2^24 bytes](https://github.com/dotnet/roslyn/issues/9852).
+When the limit is reached, the following compiler error is reported by Roslyn:
+
+```
+error CS8103: Combined length of user strings used by the program exceeds allowed limit. Try to decrease use of string literals.
+```
+
+By turning on [the feature flag](#configuration), string literals ([where possible](#eligible-string-literals)) are instead emitted as UTF-8 data into a different section of the PE file
+which does not have the same limit. The emit format is similar to [explicit u8 string literals][u8-literals].
+
+The feature is currently implemented only for C#, not VB.
+
+> [!WARNING]
+> This feature is currently experimental and can be changed or removed at any time.
+
+## Configuration
+
+The feature flag can take a non-negative integer threshold.
+Only string literals whose length (number of characters, not bytes) is over the threshold are emitted using the utf8 encoding strategy.
+If the flag is set, but no value is specified, the threshold defaults to 100.
+Specifying 0 means all string literals are considered for the feature.
```

Comment:
Should empty string literals be affected by the setting at all?

Reply:
They are not, only string literals with length *greater than* (not equal to) the threshold are affected. I will clarify that.